File: windowslib/src/main/resources/scala-native/windows/winnt/accessRights.c
#if defined(_WIN32) || defined(WIN32)
#define WIN32_LEAN_AND_MEAN
#include <windows.h>

DWORD scalanative_generic_all() { return GENERIC_ALL; }

DWORD scalanative_generic_execute() { return FILE_GENERIC_EXECUTE; }
DWORD scalanative_execute() { return FILE_EXECUTE; }
DWORD scalanative_standard_rights_execute() { return STANDARD_RIGHTS_EXECUTE; }

DWORD scalanative_generic_read() { return FILE_GENERIC_READ; }
DWORD scalanative_read_attributes() { return FILE_READ_ATTRIBUTES; }
DWORD scalanative_read_data() { return FILE_READ_DATA; }
DWORD scalanative_read_ea() { return FILE_READ_EA; }
DWORD scalanative_standard_rights_read() { return STANDARD_RIGHTS_READ; }

DWORD scalanative_generic_write() { return FILE_GENERIC_WRITE; }
DWORD scalanative_append_data() { return FILE_APPEND_DATA; }
DWORD scalanative_write_attributes() { return FILE_WRITE_ATTRIBUTES; }
DWORD scalanative_write_data() { return FILE_WRITE_DATA; }
DWORD scalanative_write_ea() { return FILE_WRITE_EA; }
DWORD scalanative_standard_rights_write() { return STANDARD_RIGHTS_WRITE; }
DWORD scalanative_synchronize() { return SYNCHRONIZE; }

#endif // defined(Win32)

File: windowslib/src/main/scala/scala/scalanative/windows/winnt/AccessRights.scala
package scala.scalanative.windows.winnt

import scala.scalanative.unsafe.*

@link("advapi32")
@extern
object AccessRights {
  @name("scalanative_generic_all")
  def FILE_GENERIC_ALL: AccessRights = extern

  // execute
  @name("scalanative_generic_execute")
  def FILE_GENERIC_EXECUTE: AccessRights = extern

  @name("scalanative_execute")
  def FILE_EXECUTE: AccessRights = extern

  @name("scalanative_read_attributes")
  def STANDARD_RIGHTS_EXECUTE: AccessRights = extern

  // read
  @name("scalanative_generic_read")
  def FILE_GENERIC_READ: AccessRights = extern

  @name("scalanative_read_attributes")
  def FILE_READ_ATTRIBUTES: AccessRights = extern

  @name("scalanative_read_data")
  def FILE_READ_DATA: AccessRights = extern

  @name("scalanative_read_ea")
  def FILE_READ_EA: AccessRights = extern

  @name("scalanative_standard_rights_read")
  def STANDARD_RIGHTS_READ: AccessRights = extern

  // write
  @name("scalanative_generic_write")
  def FILE_GENERIC_WRITE: AccessRights = extern

  @name("scalanative_append_data")
  def FILE_APPEND_DATA: AccessRights = extern

  @name("scalanative_write_attributes")
  def FILE_WRITE_ATTRIBUTES: AccessRights = extern

  @name("scalanative_write_data")
  def FILE_WRITE_DATA: AccessRights = extern

  @name("scalanative_write_ea")
  def FILE_WRITE_EA: AccessRights = extern

  @name("scalanative_standard_rights_write")
  def STANDARD_RIGHTS_WRITE: AccessRights = extern

  @name("scalanative_synchronize")
  def SYNCHRONIZE: AccessRights = extern
}
